src/sql/SqlService.ts

```diff
@@ -0,0 +1,290 @@
+
+    static readonly DEFAULT_SCHEMA: string | null = null;
+
+    constructor(
+        private readonly connectionRegistry: ConnectionRegistry,
+        private readonly serializationService: SerializationService,
+        private readonly invocationService: InvocationService,
+        private readonly connectionManager: ConnectionManager
+    ) {
+    }
+
+    /**
+     * Handles SQL execute response.
+     * @param clientMessage The response message
+     * @param res SQL result for this response
+     */
+    handleExecuteResponse(clientMessage: ClientMessage, res: SqlResultImpl): void {
+        const response = SqlExecuteCodec.decodeResponse(clientMessage);
+        if (response.error !== null) {
+            res.onExecuteError(
+                new HazelcastSqlException(
+                    response.error.originatingMemberId, response.error.code, response.error.message
+                )
+            );
+        } else {
+            res.onExecuteResponse(
+                response.rowMetadata !== null ? new SqlRowMetadataImpl(response.rowMetadata) : null,
+                response.rowPage,
+                response.updateCount
+            );
+        }
+    }
+
+    /**
+     * Validates SqlStatement
+     *
+     * @param sqlStatement
+     * @throws RangeError if validation is not successful
+     * @internal
+     */
+    static validateSqlStatement(sqlStatement: SqlStatement | null): void {
+        if (sqlStatement === null) {
+            throw new RangeError('Sql statement cannot be null');
+        }
+        tryGetString(sqlStatement.sql);
+        if (sqlStatement.hasOwnProperty('params')) // if params is provided, validate it
+            tryGetArray(sqlStatement.params);
+        if (sqlStatement.hasOwnProperty('options')) // if options is provided, validate it
+            SqlServiceImpl.validateSqlStatementOptions(sqlStatement.options);
+    }
+
+    /**
+     * Validates SqlStatementOptions
+     *
+     * @param sqlStatementOptions
+     * @throws RangeError if validation is not successful
+     * @internal
+     */
+
+    static validateSqlStatementOptions(sqlStatementOptions: SqlStatementOptions): void {
+        if (sqlStatementOptions.hasOwnProperty('schema'))
+            tryGetString(sqlStatementOptions.schema);
+
+        if (sqlStatementOptions.hasOwnProperty('timeoutMillis')) {
+            const longValue = tryGetLong(sqlStatementOptions.timeoutMillis);
+            if (longValue.lessThanOrEqual(Long.fromInt(-2))) {
+                throw new RangeError('Timeout millis cannot be less than -1');
+            }
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('cursorBufferSize') && tryGetNumber(sqlStatementOptions.cursorBufferSize) <= 0) {
+            throw new RangeError('Cursor buffer size cannot be negative');
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('expectedResultType'))
+            tryGetEnum(SqlExpectedResultType, sqlStatementOptions.expectedResultType);
```

Comment:
I think it would be good to use `{}` even if the if statement has a body containing single line to make it consistent across the code base